src/udev/udev-builtin-hwdb.c

```diff
@@ -58,15 +58,16 @@ static const char *modalias_usb(sd_device *dev, char *s, size_t size) {
                 return NULL;
         if (safe_atoux16(p, &pn) < 0)
                 return NULL;
-        snprintf(s, size, "usb:v%04Xp%04X*", vn, pn);
+        (void) sd_device_get_sysattr_value(dev, "product", &n);
+
+        snprintf(s, size, "usb:v%04Xp%04X:%s", vn, pn, n);
```

Comment:
Maybe `strna(n)`? Or
```
snprintf(s, size, "usb:v%04Xp%04X%s%s", vn, pn, n ? ":" : "", strempty(n));
```

Reply:
Good point. Updated with `strempty`.

Reply:
(I kept the ":". I think it should be there to allow easy field separation.)